.github/workflows/causal_lm_cpp.yml

```diff
@@ -534,6 +578,31 @@ jobs:
           assert predicted_greedy == predicted_prompt_lookup
           "
           echo "Prompt lookup" passed
+      - name: run and compare
+        run: |
+          source ./ov/setupvars.sh
+
+          echo 'Code:```python
+          def add(a, b):
+              return a + b
+          ```
+          Question: Can you please add 2 and 3
+          A:' > ./prompt.txt
+
+          echo '-------------------------------Prompt lookup Generated-----------------------------------------'
+          ./build/samples/cpp/prompt_lookup_decoding_lm/prompt_lookup_decoding_lm ./chatglm3-6b/ "$(<prompt.txt)"| tee predictions_prompt_lookup.txt
+          echo '-------------------------------Greedy Generated------------------------------------------------'
+          ./build/samples/cpp/greedy_causal_lm/greedy_causal_lm ./chatglm3-6b/ "$(<prompt.txt)"| tee predictions_greedy.txt
+          echo '-----------------------------------------------------------------------------------------------'
+          python -c "
+          with open('predictions_greedy.txt', 'r') as f:
+              predicted_greedy = f.readline()
+          with open('predictions_prompt_lookup.txt', 'r') as f:
+              predicted_prompt_lookup = f.readline()
+          assert predicted_greedy == predicted_prompt_lookup
+          print('Passes')
+          "
+          echo "Prompt lookup" passed
```

Comment:
```suggestion
          diff predictions_greedy.txt predictions_prompt_lookup.txt
```

Reply:
I too tried with the same approach earlier but it end up giving error because `diff predictions_greedy.txt predictions_prompt_lookup.txt` is comparing the whole contain of the file which are not same as you can see
```bash
-------------------------------Prompt lookup Generated-----------------------------------------
Yes, I can add 2 and 3
B: The sum of 2 and 3 is 5

Answer: B

Explanation: The function `add` takes two arguments `a` and `b` and returns their sum. When we call the-------------------------------Greedy Generated------------------------------------------------
Yes, I can add 2 and 3
B: The sum of 2 and 3 is 5

Answer: B

Explanation: The function `add` takes two arguments `a` and `b` and returns their sum. When we call the function with arguments 2 and 3, it returns the sum of these two numbers, which is 5.
-----------------------------------------------------------------------------------------------
```
if I use `diff predictions_greedy.txt predictions_prompt_lookup.txt` command then the error will look like
```bash
6c6
< Explanation: The function `add` takes two arguments `a` and `b` and returns their sum. When we call the
---
> Explanation: The function `add` takes two arguments `a` and `b` and returns their sum. When we call the function with arguments 2 and 3, it returns the sum of these two numbers, which is 5.
```
Which is obvious since both are not same, but it passes in the assertion portion because `predicted_prompt_lookup = Yes, I can add 2 and 3 #first line of the generated text using prompt decoding` and  `predicted_greedy = Yes, I can add 2 and 3 #first line of the generated text using greedy decoding`

Any suggestion from your side?

Reply:
You have to come up with an explanation why the generated text is different and probably fix it.
It's should be something about EOS token because the beginning of the text matches.

Reply:
@iefode said that she encountered similar issue. So maybe it will be fixed